### ISSUE TITLE
Feature requests: Asymmetric swarmplot function

### ISSUE THREAD

**Issue by sunroofgod** [mwaskom/seaborn#3554]
Hi! 

I wanted to check how open the seaborn team would be to adding an additional param to the `swarmplot()` function to plot asymmetric swarmplots as well.

For example, users could potentially call

1. with a new param `side`: "r" | "l" | "c", which represents the sides "right", "left", "center" respectively that the swarm plot would be aligned against. ("center" would be the same as how the swarmplots are currently being plotted)
```python
import seaborn as sns

tips = sns.load_dataset("tips")
sns.swarmplot(data=tips, x="day", y="total_bill", hue="day", legend=False, side="right")
```

2. perhaps in more alignment with the API of the current `violinplot()` function, a new param `split` instead could be used to reproduce the same plot as the one above. 

```python
sns.swarmplot(data=tips, x="day", y="total_bill", hue="day", legend=False, split=True)
```
Both of which would result in the following plot below:

![swarmplot_side= right](https://github.com/mwaskom/seaborn/assets/90504567/f2734cc7-e1a9-46ab-91b4-b3f59ce0928f)

(The plot above has been produced via code edits I've made to the current `swarmplot()` function wrt point 1.)

To sum it up,
- Do you think this feature could be added to seaborn?
- Would code snippets of the edits I've made be helpful for this process?
- If yes to all of the above, what else can I do to help?

Thank you so much!
